machine-runner/src/runner/runner.ts

```diff
@@ -185,34 +172,22 @@ export const createMachineRunner = <
     attemptStartFrom: { from: {}, latestEventKey: EventKey.zero },
   }
 
-  const persist: PersistFn<RegisteredEventsFactoriesTuple> = (e) => sdk.publish(tags.apply(...e))
+  const persist: PersistFn<MachineEvent.Any> = (e) => sdk.publish(tags.apply(...e))
 
-  const subscribe: SubscribeFn<RegisteredEventsFactoriesTuple> = (callback, onCompleteOrErr) =>
-    sdk.subscribeMonotonic<MachineEvent.Factory.ReduceToEvent<RegisteredEventsFactoriesTuple>>(
-      subscribeMonotonicQuery,
-      callback,
-      onCompleteOrErr,
-    )
```

Comment:
perhaps it would work better if we kept the reduced event union as a type parameter in addition to RegisteredEventsFactoriesTuple — if we then still need that one

Reply:
I'll work on the better reduction process first before reading any params that depends on any kind of reductions